### PR TITLE
pkp/pkp-lib#1092: Add HTML entity decode to stripAssocArray() for conversion of HTML to plaintext.

### PR DESCRIPTION
**Pull request by ctgraham**
Resolves https://github.com/pkp/pkp-lib/issues/1092, at least to the degree that PHP is natively able, by decoding known entities into the specified client character set.

### CODE REVIEW

File: includes/functions.inc.php [hunk 1]
```diff
@@ -328,7 +328,7 @@ function arrayClean(&$array) {
 function stripAssocArray($values) {
 	foreach ($values as $key => $value) {
 		if (is_scalar($value)) {
-			$values[$key] = strip_tags($values[$key]);
+			$values[$key] = html_entity_decode(strip_tags($values[$key]), ENT_QUOTES, Config::getVar('i18n', 'client_charset'));
```
Comment (asmecher) [inline includes/functions.inc.php]: The OAI interface is fixed to use UTF8, regardless of what the client charset is set to, per the OAI-PMH spec.

Comment (ctgraham) [inline includes/functions.inc.php]: Even better.  Will fix.

**Comment by ctgraham** [comment]
This new change is overkill.  This function is currently used only in the context of OAI-PMH, but is defined outside of the OAI-PMH context.  I felt dirty hardcoding the function to UTF-8, so now it has an unused parameter.

File: includes/functions.inc.php [hunk 2]
```diff
@@ -322,13 +322,14 @@ function arrayClean(&$array) {
...
+ * Assumes entity decoding to UTF-8 (primarily for OAI-PMH)
  * @param $values array
  * @return array the cleansed array
  */
-function stripAssocArray($values) {
+function stripAssocArray($values, $useClientCharset = false) {
 	foreach ($values as $key => $value) {
 		if (is_scalar($value)) {
-			$values[$key] = strip_tags($values[$key]);
+			$values[$key] = html_entity_decode(strip_tags($values[$key]), ENT_QUOTES, $useClientCharset ? Config::getVar('i18n', 'client_charset') : 'UTF-8');
```
Comment (mfelczak) [inline includes/functions.inc.php]: @ctgraham, we also have String::html2utf as an alternative to PHP's built-in html_entity_decode.

Comment (ctgraham) [inline includes/functions.inc.php]: Sure, I can adapt it to that.

**Comment by ctgraham** [comment]
Transitioned the default usage to the more consistent `String::html2utf()` per @mfelczak.

**Pull request merged**